R/install.R

```diff
@@ -1,75 +0,0 @@
-#' Install LaminDB
-#'
-#' Create a Python environment containing **lamindb** or install **lamindb**
-#' into an existing environment.
-#'
-#' @param ... Additional arguments passed to `reticulate::py_install()`
-#' @param envname String giving the name of the environment to install packages
-#'   into
-#' @param extra_packages A vector giving the names of additional Python packages
-#'   to install
-#' @param new_env Whether to remove any existing `virtualenv` with the same name
-#'   before creating a new one with the requested packages
-#' @param use Whether to attempt use the new environment
-#'
-#' @return `NULL`, invisibly
-#' @export
-#'
-#' @details
-#' See `vignette("setup", package = "laminr")` for further details on setting up
-#' a Python environment
-#'
-#' @examples
-#' \dontrun{
-#' install_lamindb()
-#'
-#' # Add additional packages to the environment
-#' install_lamindb(extra_packages = c("bionty", "wetlab"))
-#'
-#' # Install into a different environment
-#' install_lamindb(envvname = "your-env")
-#' }
-install_lamindb <- function(
```

Comment:
could we add this function back but generate a warning if still used? something along the lines of:

```r
install_lamindb <- function(...) {
  cli_warn("This function is deprecated. Use require_lamindb() instead.")
  require_lamindb(...)
}
```

This because the laminr interface is now supposed to be stable, so we should follow a deprecation cycle before removing parts of the interface.

Or... if I interpret correctly that the package will automatically manage package installs, we can just:


```r
install_lamindb <- function(...) {
  cli_warn("This function is deprecated as it is no longer necessary to install lamindb using this function. See the documentation at ... for more information on how package management in laminr works.")
}
```

Reply:
> This because the laminr interface is now supposed to be stable, so we should follow a deprecation cycle before removing parts of the interface.

Yes, things have to be deprecated before removed. That's very important.

Reply:
Good point, I wanted to make sure it wasn't being used anywhere but I should put it back. I was actually wondering if we should leave it there with a warning but the old functionality in case someone prefers to do things that way and/or compatibility with other packages that still use the old approach?